### DISCUSSION TITLE
How to switch to window instead of changing buffers from the tabline section of the cookbook

### DISCUSSION THREAD

**Discussion by katawful** [rebelot/heirline.nvim#188] [Q&A]
In [this](https://github.com/rebelot/heirline.nvim/blob/master/cookbook.md#bufferline) part of the cookbook, left/right click for TablineFileNameBlock switches buffers in the current window. However, I prefer the bufferlist to simply display the open windows of the tabpage, but I can't seem to figure out how to get winid's from within heirline so I can focus them when clicked. I have tried setting `minwid` in the `on_click` table of it to `self.winid`, but does not work. Suggestions?

**Comment by rebelot** [comment]
you probably want to override the buf_func parameter of make_buflist to using `vim.api.nvim_tabpage_get_win(0)`. This way you'll only see windows in the current tab page, which, in my opinion, is a bit of a waste since this information may already come from statusline/winbar
